docs/api/breaking-changes-ns.md

```diff
@@ -0,0 +1,61 @@
+# Breaking changes (NetworkService)
+
+This document describes changes to Electron APIs after migrating network code
+to NetworkService API.
+
+We don't currently have an estimate of when we will enable `NetworkService` by
+default in Electron, but as Chromium is already removing non-`NetworkService`
+code, we might switch before Electron 10.
+
+The content of this document should be moved to `breaking-changes.md` once we have
+determined when to enable `NetworkService` in Electron.
+
+## Planned Breaking API Changes
+
+### `protocol.unregisterProtocol`
+### `protocol.uninterceptProtocol`
+
+The APIs are now synchronous and the optional callback is no longer needed.
+
+```javascript
+// Deprecated
+protocol.unregisterProtocol(scheme, () => { /* ... */ })
+// Replace with
+protocol.unregisterProtocol(scheme)
```

Comment:
Do we want to return the error values for synchronous version when the optional callback support is removed, without that the api might look blind.

Reply:
Yeah there should be a way to report errors. I wonder if we should throw exceptions for errors, to match the behavior of Node.js sync APIs.

Reply:
Agreed, that seems to be the pattern thats followed in existing synchronous apis

Reply:
I'll do it in another PR.